### PR TITLE
fix(windows): context help links

### LINKED ISSUE #4558: bug(windows): Can't reach "Keyman Desktop User Guide" (context help?)
From the current Keyman for Windows 14.0.248-beta 

The context help can't be reached

### Steps
Keyman Configuration --> "Help" button in the top right (next to Display language)

image

The title should also be renamed to "Keyman for WIndows"

There is a TODO to figure out linking to the help/context/ files

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #4558.

This makes a number of fixes to get the context help working correctly
in Keyman 14:

1. Adds support for `redirect:` header in help .md files so that we can do redirects for the context content in Keyman for Windows help in the lua filters.
2. Renames Keyman for Windows help files to .htm, because that's what the .chm format is expecting, and what Delphi assumes for its .chm integration, and thus splits the html and htm lua filters.
3. Renames and makes more consistent the various context help pages.
4. Adds redirects for all pages that have sufficient content on other pages.
5. Fixes the instantiation of help in various Keyman for Windows forms.
6. Defaults to opening help from the Keyman source repo if the calling app is running in the source repo (this is just a development-side tweak that simplifies my life).

### CODE REVIEW

File: windows/src/desktop/help/context/change-hotkey.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+title: Set hotkey
+redirect: ../start/hotkey_set
```
Comment (darcywong00) [inline windows/src/desktop/help/context/change-hotkey.md:3]: Will help.keyman.com need an update to handle `redirect`?

When I copied the file locally, I get

image

File: windows/src/desktop/help/keymandesktop.hhp [hunk 2]
```diff
@@ -4,14 +4,14 @@ Compatibility=1.1 or later
 Compiled file=keymandesktop.chm
 Contents file=toc.hhc
 Default Window=Main
-Default topic=index.html
+Default topic=index.htm
 Display compile progress=Yes
 Full-text search=Yes
 Language=0x409 English (United States)
 Title=Keyman Desktop User Guide
```
Comment (darcywong00) [inline windows/src/desktop/help/keymandesktop.hhp:11]:
```suggestion
Title=Keyman for Windows User Guide
```

File: windows/src/desktop/help/context/change-hotkey.md (hunk 1 above)
Comment (mcdurdin) [reply windows/src/desktop/help/context/change-hotkey.md:3]: Fixed in https://github.com/keymanapp/keyman.com/pull/246, https://github.com/keymanapp/help.keyman.com/pull/337 and https://github.com/keymanapp/help.keyman.com/pull/338!

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 14.0.258-beta](https://keyman.com/downloads/releases/beta/14.0.258)

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.20-alpha](https://keyman.com/downloads/releases/alpha/15.0.20)

**Pull request merged**